R/utils.R

```diff
@@ -158,15 +158,22 @@ reset_lang <- function(old_lang) {
 #' @param fun A function that takes a source file and returns `lint` objects.
 #' @param name Default name of the Linter.
 #' Lints produced by the linter will be labelled with `name` by default.
+#' @param linter_level Which level of expression is the linter working with?
+#'   `"expression"` means an individual expression in `xml_parsed_content`, while `"file"` means all expressions
+#'   in the current file are available in `full_xml_parsed_content`.
+#'   `NA` means the linter will be run with both, expression-level and file-level source expressions.
```

Comment:
We should mention that `NA` is really only available for back-compatibility, and using one of the others is preferable?

Also, note from below: unless we support (logical) `NA`, we need to clarify `NA_character_` here.